src/Form/EventGroupCategoryType.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace AppBundle\Form;
+
+use AppBundle\Entity\EventCategory;
+use Doctrine\ORM\EntityRepository;
+use Symfony\Bridge\Doctrine\Form\Type\EntityType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class EventGroupCategoryType extends AbstractType
+{
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults([
+            'class' => EventCategory::class,
+            'choice_label' => 'name',
+            'query_builder' => function (EntityRepository $er) {
```

Comment:
This logic should take place in a repository imho..

how about you typehint with the specific repository, and then use a custom method of that repository ?
e.g.: 
```
'query_builder' => function (EventCategoryRepository $repository) {
    return $repository->createQueryBuilderForMyEpicFormChoices();
}
```
What do you think?

Reply:
edited :+1: